### PR TITLE
Updated xml docs for tree-based trainers.

### PR DESCRIPTION
**Pull request by shmoradims**
Updated XML documentation for tree-based trainers (FastTree, FastForest, GAM, etc). Related to #2522.

Samples to come in a separate PR.

### CODE REVIEW

File: src/Microsoft.ML.FastTree/doc.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
           The output of the ensemble produced by MART on a given instance is the sum of the tree outputs.
         </para>
         <list type='bullet'>
-          <item><description>In case of a binary classification problem, the output is converted to a probability by using some form of calibration.</description></item>
```
Comment (zeahmed) [inline src/Microsoft.ML.FastTree/doc.xml]: >case [](start = 32, length = 4)

is the deletion of word `case` by mistake?

File: src/Microsoft.ML.FastTree/doc.xml [hunk 2]
```diff
@@ -102,43 +66,29 @@
...
-      </example>
-    </example>
 
-    <member name="FastTreeTweedieRegression">
-      <summary>
-        Trains gradient boosted decision trees to fit target values using a Tweedie loss function. 
-        This learner is a generalization of Poisson, compound Poisson, and gamma regression.
-      </summary>
+    <!--  
+    The following text describes the FastForest algorithm details.
```
Comment (zeahmed) [inline src/Microsoft.ML.FastTree/doc.xml]: >FastForest [](start = 37, length = 10)

GAMs???

File: src/Microsoft.ML.FastTree/FastTreeArguments.cs [hunk 3]
```diff
@@ -594,7 +643,7 @@ public abstract class BoostedTreeOptions : TreeOptions
         public bool BestStepRankingRegressionTrees = false;
 
         /// <summary>
-        /// Should we use line search for a step size.
+        /// Determines whether to we use line search for a step size.
```
Comment (zeahmed) [inline src/Microsoft.ML.FastTree/FastTreeArguments.cs]: >to we use [](start = 31, length = 9)

`to we use` -> `to use`

File: src/Microsoft.ML.FastTree/FastTreeArguments.cs [hunk 4]
```diff
@@ -611,11 +660,17 @@ public abstract class BoostedTreeOptions : TreeOptions
         [Argument(ArgumentType.LastOccurenceWins, HelpText = "Minimum line search step size", ShortName = "minstep")]
         public Double MinimumStepSize;
 
+        /// <summary>
+        /// The type of optimizer algorithm for setting <see cref="OptimizationAlgorithm"/>.
```
Comment (zeahmed) [inline src/Microsoft.ML.FastTree/FastTreeArguments.cs]: >The type of optimizer algorithm for setting <see cref="OptimizationAlgorithm"/>. [](start = 12, length = 80)

I think only `Types of optimization algorithms.` will make more sense here.

**Comment by singlis** [comment]
>     /// The base class for all unsupervised learner inputs that support a weight column.

is it worth changing all "learner" instances to trainer in this file? 


---
Refers to: src/Microsoft.ML.Data/Training/TrainerInputBase.cs:85 in 0966dca. [](commit_id = 0966dcae3eee67625008d5153fd7f119a6b6f61c, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.Regression
+{
+    public static class FastTree
+    {
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs]: >{ [](start = 4, length = 1)

So do we need a Binary and Ranking examples for FastTree?

**Comment by singlis** [comment]
>         /// <param name="featureToInputMap">A map from the feature shape functions (as described by the binUpperBounds and BinEffects)

see crefs for binUpperBounds and binEffects. I know this is internal -- but it would bound to the variable name if they ever change.

---
Refers to: src/Microsoft.ML.FastTree/GamClassification.cs:189 in 0966dca. [](commit_id = 0966dcae3eee67625008d5153fd7f119a6b6f61c, deletion_comment = False)

File: src/Microsoft.ML.FastTree/doc.xml (hunk 1 above)
Comment (shmoradims) [reply src/Microsoft.ML.FastTree/doc.xml]: good catch

---
In reply to: [265813438](https://github.com/dotnet/machinelearning/pull/2970#discussion_r265813438) [](ancestors = 265813438)

File: src/Microsoft.ML.FastTree/doc.xml (hunk 2 above)
Comment (shmoradims) [reply src/Microsoft.ML.FastTree/doc.xml]: right, copy-paste error

---
In reply to: [265813553](https://github.com/dotnet/machinelearning/pull/2970#discussion_r265813553) [](ancestors = 265813553)

File: src/Microsoft.ML.FastTree/FastTreeArguments.cs (hunk 3 above)
Comment (shmoradims) [reply src/Microsoft.ML.FastTree/FastTreeArguments.cs]: fixed

---
In reply to: [265814516](https://github.com/dotnet/machinelearning/pull/2970#discussion_r265814516) [](ancestors = 265814516)

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs (hunk 5 above)
Comment (shmoradims) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs]: all the samples will come in my next PR. please also see my email for the suggested template for those samples. 

---
In reply to: [265818226](https://github.com/dotnet/machinelearning/pull/2970#discussion_r265818226) [](ancestors = 265818226)

**Comment by shmoradims** [comment]
>         /// <param name="featureToInputMap">A map from the feature shape functions (as described by the binUpperBounds and BinEffects)

done

---
In reply to: [473121003](https://github.com/dotnet/machinelearning/pull/2970#issuecomment-473121003) [](ancestors = 473121003)

---
Refers to: src/Microsoft.ML.FastTree/GamClassification.cs:189 in 0966dca. [](commit_id = 0966dcae3eee67625008d5153fd7f119a6b6f61c, deletion_comment = False)

**Comment by shmoradims** [comment]
>     /// The base class for all unsupervised learner inputs that support a weight column.

done

---
In reply to: [473118846](https://github.com/dotnet/machinelearning/pull/2970#issuecomment-473118846) [](ancestors = 473118846)

---
Refers to: src/Microsoft.ML.Data/Training/TrainerInputBase.cs:85 in 0966dca. [](commit_id = 0966dcae3eee67625008d5153fd7f119a6b6f61c, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            var examples = GenerateRandomDataPoints(1000);
+
+            // Convert the examples list to an IDataView object, which is consumable by ML.NET API.
+            var data = mlContext.Data.LoadFromEnumerable(examples);
+
+            // Define the trainer.
+            var pipeline = mlContext.BinaryClassification.Trainers.FastTree();
+
+            // Train the model.
+            var model = pipeline.Fit(data);
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs]: This is minimal version without prediction. I personally like to see in-memory prediction, which is what will happen immediately in production.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                yield return new DataPoint
+                {
+                    Label = label,
+                    // Create random features that are correlated with label.
+                    Features = Enumerable.Repeat(label, 50).Select(x => x + randomFloat()).ToArray()
+                };
+            }
+        }
+
+        private class DataPoint
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs]: This is really good! An example with 50 features!

**Pull request merged**